tests/conditional_formatting/conditional_formatting_plugin.test.ts

```diff
@@ -679,6 +680,275 @@ describe("conditional format", () => {
     });
   });
 
+  describe("Grid Manipulation Formulas", () => {
+    function cellIsRuleEqual(f: string): ConditionalFormatRule {
+      return {
+        type: "CellIsRule",
+        values: [f],
+        operator: "Equal",
+        style: { fillColor: "orange" },
+      };
+    }
+
+    function cellIsRuleBetween(f: string): ConditionalFormatRule {
+      return {
+        type: "CellIsRule",
+        values: [f, f],
+        operator: "Between",
+        style: { fillColor: "orange" },
+      };
+    }
+
+    function colorScaleRule(f: string): ConditionalFormatRule {
+      return {
+        type: "ColorScaleRule",
+        minimum: {
+          type: "formula",
+          value: f,
+          color: 0,
+        },
+        maximum: {
+          type: "formula",
+          value: f,
+          color: 0,
+        },
+        midpoint: {
+          type: "formula",
+          value: f,
+          color: 0,
+        },
+      };
+    }
+
+    function iconSetRule(f: string): ConditionalFormatRule {
+      return {
+        type: "IconSetRule",
+        icons: {
+          upper: "u",
+          middle: "m",
+          lower: "l",
+        },
+        upperInflectionPoint: {
+          type: "formula",
+          operator: "gt",
+          value: f,
+        },
+        lowerInflectionPoint: {
+          type: "formula",
+          operator: "gt",
+          value: f,
+        },
+      };
+    }
+
+    function formulasToCFs(formulas: string[], cfrule: any): ConditionalFormat[] {
+      return formulas
+        .map((f) => cfrule(f))
+        .map((rule) => {
+          return {
+            id: "1",
+            rule,
+            ranges: ["A1"],
+          };
+        });
+    }
+
+    const cfGenerators = [
+      ["cellIsRuleEqual", cellIsRuleEqual],
+      ["cellIsRuleBetween", cellIsRuleBetween],
+      ["colorScaleRule", colorScaleRule],
+      ["IconSetRule", iconSetRule],
+    ];
```

Comment:
you can add a better type for `cfrule`. And then you have to add `as const` otherwise Typescript infers `(string | () => ...)[]` instead of `[string, () => ...]`
```suggestion
    function formulasToCFs(formulas: string[], cfrule: (formula) => ConditionalFormatRule): ConditionalFormat[] {
      return formulas
        .map((f) => cfrule(f))
        .map((rule) => {
          return {
            id: "1",
            rule,
            ranges: ["A1"],
          };
        });
    }

    const cfGenerators = [
      ["cellIsRuleEqual", cellIsRuleEqual],
      ["cellIsRuleBetween", cellIsRuleBetween],
      ["colorScaleRule", colorScaleRule],
      ["IconSetRule", iconSetRule],
    ] as const;
```